crypto/objects/objects.txt

```diff
@@ -371,8 +375,10 @@ rsadsi 2 5		: MD5			: md5
 rsadsi 2 6		:			: hmacWithMD5
 rsadsi 2 7		:			: hmacWithSHA1
 
-member-body 156 10197 1 401 : SM3               : sm3
-member-body 156 10197 1 504 : RSA-SM3		: sm3WithRSAEncryption
+sm-scheme 301           : SM2                   : sm2
+
+sm-scheme 401           : SM3                   : sm3
+sm-scheme 504           : RSA-SM3		: sm3WithRSAEncryption
```

Comment:
This name seems to be a misunderstanding...  according to GmSSL w/ SM2, it should be 

```
sm-scheme 504		: SM2Sign-with-SHA511	: sm2sign-with-sha512
```

Is there an authoritative document for all these OIDs somewhere that can be refered to?

Reply:
@levitte [GB/T 33560-2017](http://www.gb688.cn/bzgk/gb/newGbInfo?hcno=252CF0F72A7BE339A56DEA7D774E8994) (in Chinese) describes the rather authoritative list (an informative annex within a national standard) of OIDs. 

1.2.156.10197.1.504 is "Digital Signature: based on RSA and SM3". You can also find this [OID in the SM3 Internet-Draft](https://tools.ietf.org/html/draft-sca-cfrg-sm3-00#section-8.3).

Reply:
Hmmm, so I tried the "在线预览" button in [GB/T 33560-2017](http://www.gb688.cn/bzgk/gb/newGbInfo?hcno=252CF0F72A7BE339A56DEA7D774E8994), but that didn't bring me more than this (see image)

![foo](https://user-images.githubusercontent.com/698918/34637980-949e574a-f2c2-11e7-8f15-9bd24d2b2329.png)

I have to admit that this kind of issue makes it difficult to review this PR.  That doesn't stop me from trying, though...

Thanks for the pointer to the SM3 draft...  should the GmSSL guys be notified that they use clashing OIDs?

Reply:
![41515316859_ pic_hd](https://user-images.githubusercontent.com/799430/34648151-58b95050-f3cf-11e7-8495-df4d7566d3c2.jpg)
![51515316889_ pic_hd](https://user-images.githubusercontent.com/799430/34648154-609fefe0-f3cf-11e7-9114-a2730da4f279.jpg)

@levitte I just take screen shots of the OID table in the doc you can't open, the content in that table is in Chinese and I guess that probably could guess some meanings of them...Actually we should prepare an English translate of this table online somewhere...

Reply:
Thank you @InfoHunter, that does help, I can at least understand where there are latin letters ;-) ).  Yes, I agree that it would be good with an English translation, that would help tons.

Reply:
Just did a quick translation on the above table: https://github.com/InfoHunter/misc/blob/master/GBT33560-2017-A1.md

Reply:
Thanks @InfoHunter, your translation is mighty quick 😉

Reply:
@levitte I had to use Firefox to see the document, Flash, unfortunately, is required...

Reply:
Oh...

someone might want to inform someone in CCSTC or whoever makes those pages that Flash is dead.  Mozilla turned it off in Firefox back in August last year...

Reply:
Errr, actually I think the site is not controlled by SCA/OSCCA/CCSTC, it should be maintained by other government departments. I will find opportunities to talk about this 'hard-to-get-doc-issue' (1. where to get the doc; 2. very few English resources) with some government officers. (But even you can view the doc in that site, seems you can't download it easily...)

Actually officially the SCA/OSCCA/CCSTC (can we agree on one label for these names?) is making the SM ciphers standardized internationally in ISO and Ribose (@ronaldtse and friends including BaishanCloud) is proposing the GM/T or GB/T standards into RFCs with IETF. So I guess things will be easier in the future, but pioneers always run into difficulty ;-)

Reply:
The site is maintained by the [SAC (Standardization Administration of China)](http://www.sac.gov.cn), the National Standards Body of China and the publisher of all "GB" standards.

In fact ["GB688"](http://www.gb688.cn/home) is a very new site. I doubt they are willing to see their newly made investment in Flash go up in smoke. 😉

Reply:
FYI: SCA plans to put all of their GM related standards on their website for free, but not sure about the exact time yet...